File: FindMaxMinAvgFromArray/java/SimpleMaxMinAvgWithoutArray.java
import java.text.DecimalFormat;
import java.util.Scanner;

/**
 * @author biswarup
	
	Program to find the Minimum, Maximum without array
 	
 	How to run this code ?
 	1. Compile the JAVA code in command prompt, which will create .class file (compiled version of the java code)
 	javac SimpleMaxMinAvgWithoutArray.java
 	2. Run the code by running as below:
 	java SimpleMaxMinAvgWithoutArray
	 	
	********************* ENTER INPUTS *****************************
	Enter your name::	bis
	Enter numbers you want to check::	7
	Enter number1::	23.5
	Enter number2::	12.34
	Enter number3::	-234.04
	Enter number4::	-123.56
	Enter number5::	34.56
	Enter number6::	134.7
	Enter number7::	889.5
	
	
	********************* SEE RESULTS *************************************
	
	Minimum of the numbers bis entered::	-234.04
	Maximum of the numbers bis entered::	889.5
	************************************************************************

 **/

public class SimpleMaxMinAvgWithoutArray {

	public static void main(String[] args) {
		// Gathering inputs
		System.out.println("********************* ENTER INPUTS *****************************");
		Scanner scanner = new Scanner(System.in);
		System.out.println("Enter your name::\t");
		String name = scanner.next();		
		
		System.out.println("Enter numbers you want to check::\t");
		int totalNumbers = scanner.nextInt();

		System.out.print("Enter number" + 1 + "::\t");
		double numberEntered = scanner.nextDouble();   // consider this as 0th entry
		
		double minNum = numberEntered;
		double maxNum = numberEntered;
		double sumNum = numberEntered;
		
		// For loop starting from 2nd entry to the last which is (totalNumber - 1)
		for (int i = 1; i < totalNumbers; i++) {     
			System.out.print("Enter number" + (i+1) + "::\t");
			numberEntered = scanner.nextDouble();
			
			if(numberEntered > maxNum)
				maxNum = numberEntered;
			if(numberEntered < minNum)
				minNum = numberEntered;
			
			sumNum = sumNum + numberEntered;
		}
		scanner.close();
		System.out.println("\n");
		
		// Showing results in the console
		System.out.println("********************* SEE RESULTS *************************************");
		System.out.println();
		
		System.out.println("Minimum of the numbers "+ name +" entered::\t" + minNum); 

		System.out.println("Maximum of the numbers "+ name +" entered::\t" + maxNum); 
		
		DecimalFormat format = new DecimalFormat("##.00");
		System.out.println("Average of the numbers "+ name +" entered::\t" + format.format(sumNum/totalNumbers)); 
		
		System.out.println("************************************************************************");

				
	}

}

File: FindMaxMinAvgFromArray/python/SimpleMaxMinAvgWithoutArray.py

"""
 * @author biswarup
	
	Program to find the Minimum, Maximum without array
 	
 	How to run this code ?
 	- python3 SimpleMaxMinWithoutArray.py
	 	
	********************* ENTER INPUTS *****************************
    Enter your name with quotes::
    bis
    Enter numbers you want to check::
    asd
    You entered a non-integer. Please input a correct integer!
    Enter numbers you want to check::
    4
    Enter number1:: 12
    Enter number2:: -13.4
    Enter number3:: asca
    You entered a non-numeric. Please input a correct number!
    Enter number3:: 45.6
    Enter number4:: 78

    ********************* SEE RESULTS *************************************

    Minimum of the numbers bis entered::    -13.4
    Maximum of the numbers bis entered::    78.0
    ************************************************************************

"""

def obtainInput(dataType):
    isValidInput = False
    while isValidInput == False:
        try:
            if dataType == "int":
                var = int(input())
            if dataType == "float":
                var = float(input())
            isValidInput = True
        except ValueError:
            print("Input not in correct format. Please enter correctly!")
            isValidInput = False
    return var


def main():
    print("********************* ENTER INPUTS *****************************")
    print("Enter your name with quotes::\t")
    name = input()
    print("Enter numbers you want to check::\t")	
    totalNumbers = obtainInput("int")  
        
    print("Enter number" + repr(1) + "::\t", end = "")
    numberEntered = obtainInput("float")   # consider this as 0th entry
    minNum = numberEntered
    maxNum = numberEntered
    sumNum = numberEntered

    # For loop starting from 2nd entry to the last which is (totalNumber - 1)
    for i in range(1, totalNumbers):
        print("Enter number" + repr(i+1) + "::\t", end = "")    
        numberEntered = obtainInput("float")
        
        if numberEntered > maxNum:
            maxNum = numberEntered
        if numberEntered < minNum:
            minNum = numberEntered
        
        sumNum = sumNum + numberEntered
    # End of For loop

    # Showing results in the console
    print()
    print("********************* SEE RESULTS *************************************\n")

    print("Minimum of the numbers "+ name +" entered::\t" + repr(minNum))

    print("Maximum of the numbers "+ name +" entered::\t" + repr(maxNum))

    print("Average of the numbers "+ name +" entered::\t" + repr(round(sumNum/totalNumbers,2)))

    print("************************************************************************")
# end of main()

if __name__== "__main__":
  main()